bson/src/main/org/bson/codecs/pojo/ConventionAnnotationImpl.java

```diff
@@ -31,14 +39,65 @@ public void apply(final ClassModelBuilder<?> classModelBuilder) {
             processClassAnnotation(classModelBuilder, annotation);
         }
 
-        for (final FieldModelBuilder<?> fieldModelBuilder : classModelBuilder.getFields()) {
-            for (final Annotation annotation : fieldModelBuilder.getAnnotations()) {
-                processFieldAnnotation(classModelBuilder, fieldModelBuilder, annotation);
+        for (final PropertyModelBuilder<?> propertyModelBuilder : classModelBuilder.getPropertyModelBuilders()) {
+            for (final Annotation annotation : propertyModelBuilder.getAnnotations()) {
+                processPropertyAnnotation(classModelBuilder, propertyModelBuilder, annotation);
+            }
+        }
+
+        processCreatorAnnotation(classModelBuilder);
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> void processCreatorAnnotation(final ClassModelBuilder<T> classModelBuilder) {
+        Class<T> clazz = classModelBuilder.getType();
+        CreatorExecutable<T> creatorExecutable = null;
+        for (Constructor<?> constructor : clazz.getDeclaredConstructors()) {
+            if (isPublic(constructor.getModifiers())) {
+                for (Annotation annotation : constructor.getDeclaredAnnotations()) {
+                    if (annotation.annotationType().equals(Creator.class)) {
+                        creatorExecutable = new CreatorExecutable<T>(clazz, (Constructor<T>) constructor);
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (creatorExecutable == null) {
+            for (Method method : clazz.getDeclaredMethods()) {
+                if (isStatic(method.getModifiers())) {
```

Comment:
It should also be required to return an instance of the declaring class, as per 4.5, paragraph 3.